### PR TITLE
Disallow constructing frames from a ColumnAccessor

### PR DESCRIPTION
**Pull request by shwina**
Constructing a DataFrame from a ColumnAccessor previously had unintended side-effects:

```python

In [1]: import cudf

In [2]: a = cudf.DataFrame({'a': [1, 2, 3]})

In [3]: a._data['a'].__cuda_array_interface__
Out[3]:
{'shape': (3,),
 'strides': (8,),
 'typestr': '<i8',
 'data': (140409137266688, False),
 'version': 1}

In [4]: a[['a']]
Out[4]:
   a
0  1
1  2
2  3

In [5]: a._data['a'].__cuda_array_interface__
Out[5]:
{'shape': (3,),
 'strides': (8,),
 'typestr': '<i8',
 'data': (140409137267200, False),
 'version': 1}
```

In a discussion with @galipremsagar - we decided that it's probably best not to handle `ColumnAccessor` in the frame constructors. 

* Remove special handling of `ColumnAccessor` in `Frame` constructors
* Collapse `Series.copy()` and `DataFrame.copy()` into a single `Frame.copy()`

### CODE REVIEW

**Comment by shwina** [comment]
rerun tests

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -502,6 +488,17 @@ def _from_table(cls, table, index=None):
         out._index = index
         return out
 
+    @classmethod
+    def _from_data(cls, data, index=None, columns=None):
+        out = cls()
```
Comment (shwina) [inline python/cudf/cudf/core/dataframe.py:493]:
```suggestion
        out = cls.__new__(cls)
```

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -209,8 +211,91 @@ def empty(self):
...
+        deep : bool, default True
+            Make a deep copy, including a copy of the data and the indices.
+            With ``deep=False`` neither the indices nor the data are copied.
+
+        Returns
+        -------
+        copy : Series or DataFrame
+            Object type matches caller.
+
+
```
Comment (galipremsagar) [inline python/cudf/cudf/core/frame.py:238]:
```suggestion

```

**Review by galipremsagar — APPROVED** [review]
LGTM

**Comment by galipremsagar** [comment]
rerun tests

**Comment by kkraus14** [comment]
@gpucibot merge

**Comment by kkraus14** [comment]
@gpucibot merge

**Review by kkraus14 — APPROVED** [review]

**Pull request merged**